flow360/component/simulation/framework/cached_model_base.py

```diff
@@ -1,12 +1,35 @@
 import abc
-from typing import Any, Dict
+import inspect
+from functools import wraps
+from typing import Any, Callable, Dict
 
 import pydantic as pd
 
 from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.types import TYPE_TAG_STR
 
 
 class CachedModelBase(Flow360BaseModel, metaclass=abc.ABCMeta):
+    @classmethod
+    def model_constructor(cls, func: Callable) -> Callable:
+        @classmethod
+        @wraps(func)
+        def wrapper(cls, *args, **kwargs):
+            sig = inspect.signature(func)
+            result = func(cls, *args, **kwargs)
+            defaults = {
+                k: v.default
+                for k, v in sig.parameters.items()
+                if v.default is not inspect.Parameter.empty
+            }
+            result._cached = result.__annotations__["_cached"](
+                **{**result._cached.model_dump(), **defaults, **kwargs}
```

Comment:
What happens if **result._cached.model_dump(), **defaults, **kwargs have same keyword args?

Reply:
cache will be updated. This order should be correct: existing cache < defaults < kwargs